Lib/mzscheme/argcargv.i

```diff
@@ -0,0 +1,46 @@
+/* ------------------------------------------------------------
+ * SWIG library containing argc and argv multi-argument typemaps
+ * ------------------------------------------------------------ */
+
+%typemap(in) (int ARGC, char **ARGV) {
+  $1_ltype i, len;
+  Scheme_Object **elems;
+  SWIG_contract_assert($input != (Scheme_Object *)NULL &&
+                       $input != scheme_null &&
+                       SCHEME_TYPE($input) == scheme_vector_type, "null array");
+  len = SCHEME_VEC_SIZE($input);
+  SWIG_contract_assert(len > 0, "array must contain at least 1 element");
+  $1 = len;
+  $2 = ($2_ltype) SWIG_MzScheme_Malloc((size_t)(len+1)*sizeof($*2_ltype), FUNC_NAME);
+  elems = SCHEME_VEC_ELS($input);
+  for (i = 0; i < len; i++) {
+    SWIG_contract_assert(SCHEME_TYPE(elems[i]) == scheme_char_string_type,
+                         "elements in array must be strings");
+    $2[i] = ($*2_ltype)SCHEME_STR_VAL(elems[i]);
+  }
+  $2[i] = NULL;
+}
+
+%typemap(typecheck, precedence=SWIG_TYPECHECK_STRING_ARRAY) (int ARGC, char **ARGV) {
+  if ($input != (Scheme_Object *)NULL && $input != scheme_null &&
+      SCHEME_TYPE($input) == scheme_vector_type) {
+    size_t len = SCHEME_VEC_SIZE($input);
+    if (len > 0) {
+      size_t i;
+      Scheme_Object **elems = SCHEME_VEC_ELS($input);
+      for (i = 0; i < len; i++) {
+        if (SCHEME_TYPE(elems[i]) != scheme_char_string_type) {
+          break;
+        }
+      }
+      /* All elements are strings! */
+      $1 = (i == len);
+    }
+  }
+}
+
+%typemap(freearg) (int ARGC, char **ARGV) {
+  if ($2 != NULL) {
+    SWIG_free((void *)$2);
+  }
```

Comment:
You don't need a `NULL`-check before calling `free()` or `delete`.  The ISO C and C++ standards guarantee that it's safe to call these on a NULL pointer, so it's not necessary for the calling code to also check.

I cleaned up all the cases I could find last year in 7ec2f89fe241e3aabd988d3aa6fe17e4878516dc so let's not introduce more.  (I removed the ones added by your other argcargv.i changes in fcdd63dba592fcfeb2fb60314e49eb5f2d185e15.)

Reply:
Sure.
I just followed other examples.

There are other you can clean:
Lib/octave/argcargv.i
Lib/go/argcargv.i
Lib/perl5/argcargv.i
Lib/php/argcargv.i
Lib/tcl/argcargv.i

Some you can blame me -:)